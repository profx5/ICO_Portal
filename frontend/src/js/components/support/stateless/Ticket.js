import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from "moment/moment";
import {media} from 'js/utils/media';

import iconUser from 'img/user.svg';
import iconResolved from 'img/check-green.svg';
import iconPending from 'img/icon_transit-amber.svg';


const Ticket = ({email, id, status, title, lastReplyBy, created, onClickHandler}) => {

    return (
        <StyledLink to={`/user_office/support/ticket/${id}`} onClick={onClickHandler.bind(this, id)}>
            <td callspan="1">
                <TicketInfo>
                    <div>
                        <img src={[3,4].includes(status) ? iconResolved : iconPending} alt="Ticket status"/>
                    </div>
                    <div>
                        <span>{title}</span>
                        <span>{[3,4].includes(status) ? 'Answer received' : 'Pending'}</span>
                    </div>
                </TicketInfo>
            </td>
            <TicketDetails callspan="1">
                <div>
                    <img src={iconUser} alt="User thumbnail"/>
                </div>
                <div className="hidden-xs">
                    <span>{lastReplyBy === email && 'You' || lastReplyBy || 'Unknown'}</span>
                    <span>
                        {`${lastReplyBy !== email ? "Replied" : "Message sent"} ${moment(created).format('DD MMMM YYYY')}`}
                    </span>
                </div>
            </TicketDetails>
        </StyledLink>
    )
}


Ticket.propTypes = {
    email: PropTypes.string,
    id: PropTypes.number,
    status: PropTypes.string,
    title: PropTypes.string,
    lastReplyBy: PropTypes.string,
    created: PropTypes.object,
    onClickHandler: PropTypes.func
}

export default Ticket;

const StyledLink = styled.tr`
    display: flex;
    justify-content: space-between;
    border-top: 1px solid rgb(151,151,151,.2);
    cursor: pointer;
    &:last-of-type {
        border-bottom: 1px solid rgb(151,151,151,.2);
    }
    ${media.xs} {
        padding: 15px 0;
    }
`;

const TicketInfo = styled.div`
    display: inline-flex;
    flex-flow: row nowrap;
    ${media.smPlus} {
        height: 85px;
        align-items: center;
    }
    img {
        width: 40px;
        height: auto;
        margin-right: 16px;
        ${media.xs} {
            margin-right: 10px;
            width: 32px;
        }
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        ${media.xs} {
            font-size: 14px;
        }
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: rgb(31,31,31,.4);
        letter-spacing: 0.3px;
        display: block;
        ${media.xs} {
            font-size: 12px;
        }
    }
`;

const TicketDetails = styled.td`
    display: inline-flex;
    flex-flow: row nowrap;
    ${media.smPlus} {
        height: 85px;
        align-items: center;
    }
    img {
        width: 40px;
        height: auto;
        margin-right: 13px;
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        width: 20%;
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: #1f1f1f;
        letter-spacing: 0.3px;
        opacity: 0.4;
        display: block;
    }
`;
