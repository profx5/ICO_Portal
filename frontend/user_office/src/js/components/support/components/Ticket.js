import React from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';
import moment from "moment/moment";

import iconUser from './../../../../img/user.svg';
import iconResolved from './../../../../img/check-green.svg';
import iconPending from './../../../../img/icon_transit-amber.svg';


const Ticket = ({email, id, status, title, lastReplyBy, lastReplyAt, created, onClickHandler}) => {

    return (
        <StyledTicket>
            <TicketLinkWrapper callspan="1">
                <TicketLink to={`/user_office/support/ticket/${id}`} onClick={onClickHandler.bind(this, id)}>
                    <div>
                        <img src={[3,4].includes(status) ? iconResolved : iconPending} alt="Ticket status"/>
                    </div>
                    <div>
                        <span>{title}</span>
                        <span>{[3,4].includes(status) ? 'Answer received' : 'Pending'}</span>
                    </div>
                </TicketLink>
            </TicketLinkWrapper>
            <TicketInfo callspan="1">
                <div>
                    <img src={iconUser} alt="User thumbnail"/>
                </div>
                <div>
                    <span>{lastReplyBy === email && 'You' || lastReplyBy || 'Unknown'}</span>
                    <span>
                        {`${lastReplyBy !== email ? "Replied" : "Message sent"} ${moment(created).format('DD MMMM YYYY')}`}
                    </span>
                </div>
            </TicketInfo>
        </StyledTicket>
    )
}


export default Ticket;

const StyledTicket = styled.tr`
    display: flex;
    justify-content: space-between;
    border-top: 1px solid rgb(151,151,151,.2);
    &:last-of-type {
        border-bottom: 1px solid rgb(151,151,151,.2);
    }
`;

const TicketLinkWrapper = styled.td`
`;

const TicketLink = styled(Link)`
    display: inline-flex;
    flex-flow: row nowrap;
    align-items: center;
    height: 85px;
    img {
        width: 40px;
        height: auto;
        margin-right: 16px;
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: #1f1f1f;
        letter-spacing: 0.3px;
        opacity: 0.4;
        display: block;
    }
`;

const TicketInfo = styled.td`
    display: inline-flex;
    flex-flow: row nowrap;
    align-items: center;
    height: 85px;

    img {
        width: 40px;
        height: auto;
        margin-right: 13px;
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        width: 20%;
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: #1f1f1f;
        letter-spacing: 0.3px;
        opacity: 0.4;
        display: block;
    }
    div:nth-of-type(1) {
        
    }
    div:nth-of-type(2) {

    }
`;
