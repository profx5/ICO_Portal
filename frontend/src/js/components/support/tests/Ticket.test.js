import React from 'react';
import {configure, shallow} from 'enzyme';
import jest from 'jest-mock';
import Adapter from 'enzyme-adapter-react-16';

import Ticket from 'js/components/support/stateless/Ticket';


configure({ adapter: new Adapter() });

it('should render stateless Ticket', () => {
    const props = {
        onClickHandler: jest.fn()
    }
    const component = shallow(<Ticket {...props}/>);
    expect(component).toMatchSnapshot();
})
