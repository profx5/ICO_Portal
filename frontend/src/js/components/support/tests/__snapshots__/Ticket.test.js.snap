// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless Ticket 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ticket
    onClickHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <td
          callspan="1"
        >
          <ForwardRef>
            <div>
              <img
                alt="Ticket status"
                src="icon_transit-amber.svg"
              />
            </div>
            <div>
              <span>
                
              </span>
              <span>
                Pending
              </span>
            </div>
          </ForwardRef>
        </td>,
        <ForwardRef
          callspan="1"
        >
          <div>
            <img
              alt="User thumbnail"
              src="user.svg"
            />
          </div>
          <div
            className="hidden-xs"
          >
            <span>
              You
            </span>
            <span>
              Message sent Invalid date
            </span>
          </div>
        </ForwardRef>,
      ],
      "onClick": [Function],
      "to": "/user_office/support/ticket/null",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "callspan": "1",
          "children": <ForwardRef>
            <div>
              <img
                alt="Ticket status"
                src="icon_transit-amber.svg"
              />
            </div>
            <div>
              <span>
                
              </span>
              <span>
                Pending
              </span>
            </div>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div>
                <img
                  alt="Ticket status"
                  src="icon_transit-amber.svg"
                />
              </div>,
              <div>
                <span>
                  
                </span>
                <span>
                  Pending
                </span>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Ticket status"
                  src="icon_transit-amber.svg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Ticket status",
                  "src": "icon_transit-amber.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    
                  </span>,
                  <span>
                    Pending
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pending",
                  },
                  "ref": null,
                  "rendered": "Pending",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: inline-flex;
    flex-flow: row nowrap;
    ",
                "@media (min-width: 768px)",
                " {
        height: 85px;
        align-items: center;
    }
    img {
        width: 40px;
        height: auto;
        margin-right: 16px;
        ",
                "@media (max-width: 767px)",
                " {
            margin-right: 10px;
            width: 32px;
        }
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        ",
                "@media (max-width: 767px)",
                " {
            font-size: 14px;
        }
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: rgb(31,31,31,.4);
        letter-spacing: 0.3px;
        display: block;
        ",
                "@media (max-width: 767px)",
                " {
            font-size: 12px;
        }
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "callspan": "1",
          "children": Array [
            <div>
              <img
                alt="User thumbnail"
                src="user.svg"
              />
            </div>,
            <div
              className="hidden-xs"
            >
              <span>
                You
              </span>
              <span>
                Message sent Invalid date
              </span>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="User thumbnail"
                src="user.svg"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "User thumbnail",
                "src": "user.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  You
                </span>,
                <span>
                  Message sent Invalid date
                </span>,
              ],
              "className": "hidden-xs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "You",
                },
                "ref": null,
                "rendered": "You",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Message sent Invalid date",
                },
                "ref": null,
                "rendered": "Message sent Invalid date",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    display: inline-flex;
    flex-flow: row nowrap;
    ",
              "@media (min-width: 768px)",
              " {
        height: 85px;
        align-items: center;
    }
    img {
        width: 40px;
        height: auto;
        margin-right: 13px;
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        width: 20%;
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: #1f1f1f;
        letter-spacing: 0.3px;
        opacity: 0.4;
        display: block;
    }
",
            ],
          },
          "displayName": "styled.td",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "td",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: flex;
    justify-content: space-between;
    border-top: 1px solid rgb(151,151,151,.2);
    cursor: pointer;
    &:last-of-type {
        border-bottom: 1px solid rgb(151,151,151,.2);
    }
    ",
          "@media (max-width: 767px)",
          " {
        padding: 15px 0;
    }
",
        ],
      },
      "displayName": "styled.tr",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "tr",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <td
            callspan="1"
          >
            <ForwardRef>
              <div>
                <img
                  alt="Ticket status"
                  src="icon_transit-amber.svg"
                />
              </div>
              <div>
                <span>
                  
                </span>
                <span>
                  Pending
                </span>
              </div>
            </ForwardRef>
          </td>,
          <ForwardRef
            callspan="1"
          >
            <div>
              <img
                alt="User thumbnail"
                src="user.svg"
              />
            </div>
            <div
              className="hidden-xs"
            >
              <span>
                You
              </span>
              <span>
                Message sent Invalid date
              </span>
            </div>
          </ForwardRef>,
        ],
        "onClick": [Function],
        "to": "/user_office/support/ticket/null",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "callspan": "1",
            "children": <ForwardRef>
              <div>
                <img
                  alt="Ticket status"
                  src="icon_transit-amber.svg"
                />
              </div>
              <div>
                <span>
                  
                </span>
                <span>
                  Pending
                </span>
              </div>
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div>
                  <img
                    alt="Ticket status"
                    src="icon_transit-amber.svg"
                  />
                </div>,
                <div>
                  <span>
                    
                  </span>
                  <span>
                    Pending
                  </span>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Ticket status"
                    src="icon_transit-amber.svg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Ticket status",
                    "src": "icon_transit-amber.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      
                    </span>,
                    <span>
                      Pending
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Pending",
                    },
                    "ref": null,
                    "rendered": "Pending",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    display: inline-flex;
    flex-flow: row nowrap;
    ",
                  "@media (min-width: 768px)",
                  " {
        height: 85px;
        align-items: center;
    }
    img {
        width: 40px;
        height: auto;
        margin-right: 16px;
        ",
                  "@media (max-width: 767px)",
                  " {
            margin-right: 10px;
            width: 32px;
        }
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        ",
                  "@media (max-width: 767px)",
                  " {
            font-size: 14px;
        }
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: rgb(31,31,31,.4);
        letter-spacing: 0.3px;
        display: block;
        ",
                  "@media (max-width: 767px)",
                  " {
            font-size: 12px;
        }
    }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "callspan": "1",
            "children": Array [
              <div>
                <img
                  alt="User thumbnail"
                  src="user.svg"
                />
              </div>,
              <div
                className="hidden-xs"
              >
                <span>
                  You
                </span>
                <span>
                  Message sent Invalid date
                </span>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="User thumbnail"
                  src="user.svg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "User thumbnail",
                  "src": "user.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    You
                  </span>,
                  <span>
                    Message sent Invalid date
                  </span>,
                ],
                "className": "hidden-xs",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "You",
                  },
                  "ref": null,
                  "rendered": "You",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Message sent Invalid date",
                  },
                  "ref": null,
                  "rendered": "Message sent Invalid date",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    display: inline-flex;
    flex-flow: row nowrap;
    ",
                "@media (min-width: 768px)",
                " {
        height: 85px;
        align-items: center;
    }
    img {
        width: 40px;
        height: auto;
        margin-right: 13px;
    }
    span:nth-of-type(1) {
        color: #000000;
        font-weight: 500;
        display: block;
        margin-bottom: 3px;
        width: 20%;
    }
    span:nth-of-type(2) {
        font-size: 14px;
        color: #1f1f1f;
        letter-spacing: 0.3px;
        opacity: 0.4;
        display: block;
    }
",
              ],
            },
            "displayName": "styled.td",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "td",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: flex;
    justify-content: space-between;
    border-top: 1px solid rgb(151,151,151,.2);
    cursor: pointer;
    &:last-of-type {
        border-bottom: 1px solid rgb(151,151,151,.2);
    }
    ",
            "@media (max-width: 767px)",
            " {
        padding: 15px 0;
    }
",
          ],
        },
        "displayName": "styled.tr",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "tr",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
